### DISCUSSION TITLE
HRV: Interpolation rate (different from Kubios?)

### DISCUSSION THREAD

**Discussion by Borthakur123** [neuropsychology/NeuroKit#891] [Q&A] [answered]
I have some queries related to your choice of interpolation rate and the benchmarking of your algorithm. Specifically, could you explain the rationale behind choosing an interpolation rate of 100 Hz as opposed to the 4 Hz rate that other researchers commonly use? Are there any supporting references you can share to justify this choice? Additionally, have you conducted any comparisons between NeuroKit2's outputs and those from other established software like Kubios or HRVAS?"

**Comment by DominiqueMakowski** [comment]
> I have some queries related to your choice of interpolation rate and the benchmarking of your algorithm

I'm assuming you mean for HRV computation. AFAIK there are no references that can justify it (I think I remember an issue where we talked about that, and I wonder if @danibene didn't come out with some plots to show a comparison - but perhaps it was on another topic).

 > have you conducted any comparisons between NeuroKit2's outputs and those from other established software like Kubios or HRVAS?

Not really, but that's something we'd be very happy to see/help making happen, if that's something you'd be interested in.

**Reply by danibene** [reply, marked as answer]
> I'm assuming you mean for HRV computation. AFAIK there are no references that can justify it (I think I remember an issue where we talked about that, and I wonder if @danibene didn't come out with some plots to show a comparison - but perhaps it was on another topic).

Yes, here is the link to a comment in the issue where we discussed that, where I link to a notebook with some plots: https://github.com/neuropsychology/NeuroKit/issues/671#issuecomment-1177014421

And here is the PR where we changed the interpolation rate from 1000 Hz to 100 Hz: https://github.com/neuropsychology/NeuroKit/pull/680#issuecomment-1226628839

**Reply by Borthakur123** [reply]
Thanks, I am assuming interpolating the RR signal to 100 Hz instead of 4 Hz is not going to adversely affect my HRV measures and is actually a better approach?

**Reply by martinfrasch** [reply]
Yes, in general, a higher resampling rate (assuming your raw signal was digitized at 1kHz or such) is better than lower one, but as other outlined, few systematic studies exist on the subject as to the effect on the HRV metrics. In fact, resampling rates higher than 4-10 Hz will affect your computational burden, obviously, so you need to weigh this against the other concerns. TL"DR: more is not always better; I would say, resampling at 10-20 Hz is probably ok in most cases with the caveats above. 
Also, make sure you use the same settings throughout your pipeline. 
I hope this helps. Appreciate other insights on this!
